packages/webpack-cli/lib/utils/Compiler.js

```diff
@@ -117,6 +117,11 @@ class Compiler {
             return interactive(options, outputOptions);
         }
 
+        //warn the user if bail and watch both are used together
+        if (!this.compiler.compilers && this.compiler.options.bail && this.compiler.watch) {
+            logger.warn('You are using "bail" with "watch". "bail" will still exit webpack when the first error is found.');
+        }
```

Comment:
`this.compiler` can be array when you use multi compiler mode, we need handle this